Mage/src/main/java/mage/abilities/common/DrawCardTriggeredAbility.java

```diff
@@ -109,21 +111,21 @@ public void watch(GameEvent event, Game game) {
         if (event.getType() != GameEvent.EventType.DREW_CARD) {
             return;
         }
-        if (drewOnce.contains(event.getPlayerId())) {
-            secondDrawMap.putIfAbsent(event.getPlayerId(), event.getId());
-        } else {
-            drewOnce.add(event.getPlayerId());
+        if (!drawMap.containsKey(event.getPlayerId())) {
+            drawMap.putIfAbsent(event.getPlayerId(), new ArrayList<>());
         }
+        drawMap.get(event.getPlayerId()).add(event.getId());
     }
 
     @Override
     public void reset() {
         super.reset();
-        drewOnce.clear();
-        secondDrawMap.clear();
+        drawMap.clear();
     }
-
-    static boolean checkEvent(UUID playerId, GameEvent event, Game game) {
-        return event.getId().equals(game.getState().getWatcher(DrawSecondCardWatcher.class).secondDrawMap.getOrDefault(playerId, null));
+    
+    static boolean checkEvent(UUID playerId, GameEvent event, Game game, int cardNumber) {
```

Comment:
Since we don't have a reference to the watcher in the ability, we are forced to call this static method.
Instead, if we keep a reference to the watcher in the ability, we can make this method a non-static public method, and we can avoid passing the Game parameter and avoid the getWatcher call.

Since every card has its own DrawCardTriggeredAbility instance, having a reference to the watcher in the ability should be safe.

Reply:
I think you're misunderstanding how watchers work, if a watcher is scoped to the game then there's only one instance of it and it has to be accessed through the game state. The watcher that gets instantiated with a given instance of the ability isn't the one that necessarily is being used by the game state.

Reply:
You are right, I am not completely understanding the architecture.
Anyway, is the PR ok the way it is right now?

Reply:
would you be able to write some tests for this?

Reply:
@theelk801 done.
The DrawCardTriggeredAbility is working properly, you can doublecheck the AlandraSkyDreamerTest